mtp_api/apps/transaction/tests/test_cashbook_views.py

```diff
@@ -36,800 +34,584 @@ def _get_authorised_user(self):
+            'status': TRANSACTION_STATUS.AVAILABLE
+        })
+
+    def test_WITH_status_locked_DEFAULT_prison_AND_user(self):
+        self._test_response_with_filters(filters={
+            'status': TRANSACTION_STATUS.LOCKED
+        })
+
+    def test_WITH_status_credited_DEFAULT_prison_AND_user(self):
+        self._test_response_with_filters(filters={
+            'status': TRANSACTION_STATUS.CREDITED
+        })
+
+    def test_WITH_status_available_AND_prison_DEFAULT_user(self):
+        self._test_response_with_filters(filters={
+            'prison': self.prisons[0].pk
+        })
+
+    def test_WITH_status_locked_AND_prison_DEFAULT_user(self):
+        self._test_response_with_filters(filters={
+            'status': TRANSACTION_STATUS.LOCKED,
+            'prison': self.prisons[0].pk
+        })
+
+    def test_WITH_status_credited_AND_prison_DEFAULT_user(self):
+        self._test_response_with_filters(filters={
+            'status': TRANSACTION_STATUS.CREDITED,
+            'prison': self.prisons[0].pk
+        })
+
+    def test_WITH_status_available_AND_user_DEFAULT_prison(self):
+        self._test_response_with_filters(filters={
+            'user': self.prison_clerks[1].pk
+        })
+
+    def test_WITH_status_locked_AND_user_DEFAULT_prison(self):
+        self._test_response_with_filters(filters={
+            'status': TRANSACTION_STATUS.LOCKED,
+            'user': self.prison_clerks[1].pk
+        })
+
+    def test_WITH_status_credited_AND_user_DEFAULT_prison(self):
+        self._test_response_with_filters(filters={
+            'status': TRANSACTION_STATUS.CREDITED,
+            'user': self.prison_clerks[1].pk
+        })
+
+    def test_WITH_status_available_AND_prison_AND_user(self):
+        self._test_response_with_filters(filters={
+            'prison': self.prisons[0].pk,
+            'user': self.prison_clerks[1].pk
+        })
+
+    def test_WITH_status_locked_AND_prison_AND_user(self):
+        self._test_response_with_filters(filters={
+            'status': TRANSACTION_STATUS.LOCKED,
+            'prison': self.prisons[0].pk,
+            'user': self.prison_clerks[1].pk
+        })
+
+    def test_WITH_status_credited_AND_prison_AND_user(self):
+        self._test_response_with_filters(filters={
+            'status': TRANSACTION_STATUS.CREDITED,
+            'prison': self.prisons[0].pk,
+            'user': self.prison_clerks[1].pk
+        })
+
+    def test_WITH_prison_DEFAULT_status_user(self):
+        self._test_response_with_filters(filters={
+            'prison': self.prisons[0].pk
+        })
+
+    def test_WITH_prison_AND_user_DEFAULT_status(self):
+        self._test_response_with_filters(filters={
+            'prison': self.prisons[0].pk,
+            'user': self.prison_clerks[1].pk
+        })
+
+    def test_WITH_user_DEFAULT_status_prison(self):
+        self._test_response_with_filters(filters={
+            'user': self.prison_clerks[1].pk
+        })
```

Comment:
I actually like it like this as you can see the differences between them easily and they are basically the same + function param which shows you exactly the inputs we are using for the tests.

Splitting them in different classes with just one function in them would be messy for me.
Don't you think?